integration-tests/cstor-pool/cstor_integration_test.go

```diff
@@ -0,0 +1,173 @@
+	apis "github.com/openebs/CITF/pkg/apis/openebs.io/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// TestIntegrationCstorPool function instantiate the cstor pool test suite.
+func TestIntegrationCstorPool(t *testing.T) {
+	// RegisterFailHandler is used to register failed test cases and produce readable output.
+	RegisterFailHandler(Fail)
+	// RunSpecs runs all the test cases in the suite.
+	RunSpecs(t, "Cstor pool integration test suite")
+}
+
+// Create an instance of CITF to use the inbuilt functions that will help
+// communicating with the kube-apiserver.
+var citfInstance, err = citf.NewCITF(&citfoptions.CreateOptions{
+	// K8SInclude is true to get the kube-config from the machine where the suite is running.
+	// Kube-config is a config file that establishes communication to the k8s cluster.
+	K8SInclude: true,
+})
+
+// ToDo: Set up cluster environment before runninng all test cases ( i.e. BeforeSuite)
+// The environment set up by BeforeSuite is going to persist for all
+// the test cases under run
+
+//var _ = BeforeSuite(func() {
+//	//var err error
+//	//
+//	//Expect(err).NotTo(HaveOccurred())
+//})
+
+// ToDo: Set up tear down of cluster environment ( i.e Aftersuite)
+
+// ToDo: Set up cluster environment before every test cases that will be run (i.e. preRunHook)
+// ToDo: Reset cluster environment after every test cases that will be run  ( i.e postRunHook)
+var _ = Describe("Integration Test", func() {
+	// Test Case #1 (sparse-striped-auto-spc). Type : Positive
+	When("We apply sparse-striped-auto spc yaml with maxPool count euqal to 3 on a k8s cluster having at least 3 capable node", func() {
+		It("pool resources count should be 3 with no error and online status", func() {
+			// TODO: Create a generic util function in utils.go to convert yaml into go object.
+			// ToDo: More POC regarding this util converter function.
+			// Functions generic to both cstor-pool and cstor-vol should go inside common directory
+
+			// 1.Read SPC yaml form a file.
+			// 2.Convert SPC yaml to json.
+			// 3.Marshall json to SPC go object.
+
+			// Create a storage pool claim object
+			spcObject := &apis.StoragePoolClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "disk-claim-auto",
+				},
+				Spec: apis.StoragePoolClaimSpec{
+					Name:     "sparse-claim-auto",
+					Type:     "sparse",
+					MaxPools: 3,
+					PoolSpec: apis.CStorPoolAttr{
+						PoolType: "striped",
+					},
+				},
+			}
+			// Call CITF to create StoragePoolClaim in k8s.
+			spcGot, err := citfInstance.K8S.CreateStoragePoolClaim(spcObject)
+			Expect(err).To(BeNil())
+			// We expect nil error.
+
+			// We expect 3 cstorPool objects.
+			var maxRetry int
+			var cspCount int
+			maxRetry = 10
+			for i := 0; i < maxRetry; i++ {
+				cspCount, err = getCstorPoolCount(spcGot.Name, citfInstance)
+				if err != nil {
+					break
+				}
+				if cspCount == 3 {
+					break
+				}
+				time.Sleep(time.Second * 5)
+				maxRetry--
```

Comment:
Why this line? If we are iterating over i from 0 to maxRetry then this line might not be needed

Reply:
Same applies to all code.

Reply:
Yes...